### PR TITLE
 bnxt_re/lib: Enable low latency push support

### PR DESCRIPTION
**Pull request by selvintxavier**
Enable Low latency push support for Broadcom Adapters.

### CODE REVIEW

File: providers/bnxt_re/db.c [hunk 1]
```diff
@@ -112,3 +113,147 @@ void bnxt_re_ring_cq_arm_db(struct bnxt_re_cq *cq, uint8_t aflag)
...
+	udma_to_device_barrier();
+	mmio_write64((__u64 *)pbuf->ucdb, key);
+}
+
+static inline void __bnxt_re_ring_pend_db(__u64 *ucdb, __u64 key,
+					  struct  bnxt_re_qp *qp)
+{
+	struct bnxt_re_db_hdr hdr;
+
+	bnxt_re_init_db_hdr(&hdr, *qp->jsqq->hwque->dbtail, qp->qpid, BNXT_RE_QUE_TYPE_SQ);
```
Comment (HelloTcc) [inline providers/bnxt_re/db.c:139]: The hdr here seems not used。

**Comment by selvintxavier** [comment]
Is there a way to avoid this sparse error?
/__w/1/s/providers/bnxt_re/verbs.c:1691:25: error: typename in expression
/__w/1/s/providers/bnxt_re/verbs.c:1691:25: error: undefined identifier '__attribute__'
##[error]Bash exited with code '1'.

This is an existing code already use this directive. I added an element to bnxt_re_pd_resp structure. 

struct bnxt_re_pd_resp {
        __u32 pdid;
        __u32 dpi;
        __u64 dbr;
        __u64 comp_mask;
} __attribute__((packed, aligned(4)));

If i remove the attribute, its hitting the static assert while size checking for IB_USER_VERBS_CMD_ALLOC_PD

**Comment by selvintxavier** [comment]
> Is there a way to avoid this sparse error? /__w/1/s/providers/bnxt_re/verbs.c:1691:25: error: typename in expression /__w/1/s/providers/bnxt_re/verbs.c:1691:25: error: undefined identifier '**attribute**' ##[error]Bash exited with code '1'.
> 
> This is an existing code already use this directive. I added an element to bnxt_re_pd_resp structure.
> 
> struct bnxt_re_pd_resp { __u32 pdid; __u32 dpi; __u64 dbr; __u64 comp_mask; } **attribute**((packed, aligned(4)));
> 
> If i remove the attribute, its hitting the static assert while size checking for IB_USER_VERBS_CMD_ALLOC_PD

Ignore this request. the sparse error was introduced by the second patch of he series and i fixed the patch.

File: providers/bnxt_re/db.c [hunk 2]
```diff
@@ -112,3 +113,138 @@ void bnxt_re_ring_cq_arm_db(struct bnxt_re_cq *cq, uint8_t aflag)
 	bnxt_re_init_db_hdr(&hdr, cq->cqq.head, cq->cqid, aflag);
 	bnxt_re_ring_db(cq->udpi, &hdr);
 }
+
+void bnxt_re_ring_pstart_db(struct bnxt_re_qp *qp,
+			    struct bnxt_re_push_buffer *pbuf)
+{
+	__u64 key;
```
Comment (jgunthorpe) [inline providers/bnxt_re/db.c:120]: These should be uint64_t

Comment (jgunthorpe) [reply providers/bnxt_re/db.c:120]: Everywhere

**Review by jgunthorpe — COMMENTED** [review]
Update the kernel headers commit as well

**Comment by selvintxavier** [comment]
compile tests are failing with login issue.
##[warning]Docker login failed with exit code 1, back off 00:00:10 seconds before retry.
##[error]Docker login fail with exit code 1

**Comment by selvintxavier** [comment]
Hi Jason/Leon, 
   Can this be merged?
Thanks,
Selvin

**Pull request merged**